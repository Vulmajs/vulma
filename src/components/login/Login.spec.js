import { shallow, mount } from '@vue/test-utils'
import VInput from '@components/input'
import VIcon from '@components/icon'
import VLogin from '@components/login'

describe('VLogin', () => {
    it('render correctly', () => {
        const wrapper = shallow(VLogin)

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('is vue instance', () => {
        const wrapper = shallow(VLogin)
        expect(wrapper.name()).toBe('VLogin')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('renders input element by default', () => {
        const wrapper = shallow(VInput)

        expect(wrapper.contains('input')).toBeTruthy()
        expect(wrapper.classes()).toContain('control')
    })

    it('displays the icon when the icon property is true', () => {
        const wrapper = shallow(VInput, {
            propsData: { icon: 'magnify' }
        })
        const target = wrapper.find(VIcon)

        expect(target.exists()).toBeTruthy()
    })

    it('render field password and username', () => {
        const wrapper = mount(VLogin)

        const username = wrapper.find('#username')
        expect(username.exists()).toBeTruthy()
        expect(username.attributes().type).toBe('text')

        const password = wrapper.find('#password')
        expect(password.exists()).toBeTruthy()
        expect(password.attributes().type).toBe('password')
    })

    it('render the placeholder and readonly attribute when passed', () => {
        const wrapper = shallow(VInput, {
            attrs: { placeholder: 'Awesome!', readonly: true }
        })
        const target = wrapper.find('input')

        expect(target.element.getAttribute('placeholder')).toBe('Awesome!')
        expect(target.element.getAttribute('readonly')).toBe('readonly')
    })

    it('expands input when expanded property is passed', () => {
        const wrapper = shallow(VInput, {
            propsData: { expanded: true }
        })

        expect(wrapper.classes()).toContain('is-expanded')
    })
})
