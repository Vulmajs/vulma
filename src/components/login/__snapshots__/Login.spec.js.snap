// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VLogin render correctly 1`] = `
<div class="control is-clearfix">
    <!---->
    <!---->
    <!---->
    <div class="has-text-centered">
        <a class="login-button button is-vcentered is-primary">Login</a>
    </div>
    <div class="has-text-centered">
        <!---->
        <!---->
    </div>
</div>
`;
